CRM/Contribute/BAO/Contribution.php

```diff
@@ -5005,11 +5008,17 @@ public static function checkLineItems(&$params) {
         $lineItemAmount += $item['line_total'];
       }
     }
+
     if (!isset($totalAmount)) {
       $params['total_amount'] = $lineItemAmount;
     }
-    elseif ($totalAmount != $lineItemAmount) {
-      throw new API_Exception("Line item total doesn't match with total amount.");
+    else {
+      $difference = abs($totalAmount - $lineItemAmount);
```

Comment:
This seems to create another place that would need to be refactored in order to import transactions with more than one line item. Could you put in a TODO, or create a separate PR with some logic to handle the possibility that there might be additional $lineItems?

Reply:
@JoeMurray, this fix was done specifically for creating contributions with several line items. The problem actually presents itself only when adding a contribution with several line items, as it is only then when the total amount given can be different from the sum of all line items in the contribution, due to floating point precision representation problems. If you look at the test, you'll see it has several line items for a single contribution, and that the sum of all line items is equal to the total amount, at least by human standards. Not so for the machine, though. 😃